python/ray/tune/durable_trainable.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from ray.tune.trainable import Trainable
+from ray.tune.syncer import get_cloud_sync_client
+
+
+class DurableTrainable(Trainable):
+    """A fault-tolerant Trainable.
```

Comment:
Why not make this a flag enabled feature of trainable?

Better yet, Tune can do the syncing after the trainable has written to the local file so we don't need this class at all and the sync code lives on tune lib and not the trainable class.

Reply:
Hm, maybe it's more reasonable to just do the TrainableV2 abstraction then...

Reply:
Though, TrainableV2 will block this PR further. We can mark this as experimental and merge as a separate Trainable so our BAIR users can try it out.

Then expose it better once  we implement the TrainableV2 abstraction https://github.com/ray-project/ray/issues/6417.